### PR TITLE
Use common function (scriptblock) for all tasks

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

### Changed
- Added a public alias `Set-SamplerTaskVariable` that points to the script
  `Set-SamplerTaskVariable.ps1` in the tasks folder. This alias is used to
  dot-source task variables for re-use over multiple build tasks.

This PR also fixes so that tasks folder is part of the code coverage. To get an accurate view of the code coverage.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [x] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [x] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/273)

### CODE REVIEW

**Review by gaelcolas — COMMENTED** [review]
This looks great and avoid a lot of duplication.
We can discuss the names but that's a minor point.

It'd be great if we could make it a bit cleaner than `. (Get-Command -Name 'Set-TaskScriptVariable').ScriptBlock -IsBuild`, maybe by creating an alias within build.ps1 once sampler is loaded that does that. But that's ok to look into it later.

> *[Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/273#-:-MYEC0EX0q6OYHgADwfc:b4n7b95)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @johlju)

___
*[.build/tasks/Build-Module.ModuleBuilder.build.ps1, line 43 at r1](https://reviewable.io/reviews/gaelcolas/sampler/273#-MYEBTds2lKle0PGhEfu:-MYEBTds2lKle0PGhEfv:b63njig) ([raw file](https://github.com/gaelcolas/sampler/blob/3b1cc349e6b2596a621f7e9d275ad3ce2e3bb372/.build/tasks/Build-Module.ModuleBuilder.build.ps1#L43)):*
> ```PowerShell
> Task Build_ModuleOutput_ModuleBuilder {
>     # Get the public command Set-TaskScriptVariable and dot-source its scriptblock.
>     . (Get-Command -Name 'Set-TaskScriptVariable').ScriptBlock -IsBuild
> ```

maybe -IsNewBuild or -AsNewBuild
___
*[Sampler/Public/Set-TaskScriptVariable.ps1, line 59 at r1](https://reviewable.io/reviews/gaelcolas/sampler/273#-MYEAzR5CpWZJtJb2fx-:-MYEAzR5CpWZJtJb2fx0:b-e88pjn) ([raw file](https://github.com/gaelcolas/sampler/blob/3b1cc349e6b2596a621f7e9d275ad3ce2e3bb372/Sampler/Public/Set-TaskScriptVariable.ps1#L59)):*
> ```PowerShell
> 
> #>
> function Set-TaskScriptVariable
> ```

Maybe Get-SamplerTaskVariable
Or potentially Set-SamplerTaskVariable...

**Review by johlju — COMMENTED** [review]
I can't get an Alias to work with dot-sourcing (or  a way to use`Invoke-Command` either) :confused: 

> *[Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/273#-:-MYF-9Xm826LezIxxmW-:b-kc5ko2)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @gaelcolas)

___
*[.build/tasks/Build-Module.ModuleBuilder.build.ps1, line 43 at r1](https://reviewable.io/reviews/gaelcolas/sampler/273#-MYEBTds2lKle0PGhEfu:-MYEyePh34L4nYg8Agx9:b-am53bv) ([raw file](https://github.com/gaelcolas/sampler/blob/3b1cc349e6b2596a621f7e9d275ad3ce2e3bb372/.build/tasks/Build-Module.ModuleBuilder.build.ps1#L43)):*
<details><summary><i>Previously, gaelcolas (Gael) wrote…</i></summary><blockquote>

maybe -IsNewBuild or -AsNewBuild
</blockquote></details>

Done. Was looking for a better name yesterday... `AsNewBuild` was what I was looking for, but couldn't come up with. :smile:  
___
*[Sampler/Public/Set-TaskScriptVariable.ps1, line 59 at r1](https://reviewable.io/reviews/gaelcolas/sampler/273#-MYEAzR5CpWZJtJb2fx-:-MYEytVLDQhcMPVrg-i7:b-h4081p) ([raw file](https://github.com/gaelcolas/sampler/blob/3b1cc349e6b2596a621f7e9d275ad3ce2e3bb372/Sampler/Public/Set-TaskScriptVariable.ps1#L59)):*
<details><summary><i>Previously, gaelcolas (Gael) wrote…</i></summary><blockquote>

Maybe Get-SamplerTaskVariable
Or potentially Set-SamplerTaskVariable...
</blockquote></details>

Done. Used `Set-SamplerTaskVariable` for now - feels more like it sets (and potentially overwrites) rather than getting variables. But `Set` doesn't really feel right either - more like we `Restore` or `Resolve` variable values. Maybe `Resolve-SamplerTaskVariable` is better. :thinking:

**Review by gaelcolas — APPROVED** [review]
Yeah, nvm. It's already an improvement.

> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/273#-:-MYFYd8-CNual0Ha9Ar3:b-ovbayj)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @johlju)

**Comment by johlju** [comment]
I will add more tests and fix the other tasks the same way before merging. But tomorrow I think.

**Comment by johlju** [comment]
This does not work in another repo than Sampler. Probably due to Sampler is dot-sourcing the functions into the right scope.

**Comment by johlju** [comment]
@gaelcolas I fixed 98% coverage in the the script file `Set-SamplerTaskVariable.ps1`.

File: build.ps1 [hunk 1]
```diff
@@ -270,7 +270,7 @@ process
         }
 
         # Loading Build Tasks defined in the .build/ folder (will override the ones imported above if same task name).
-        Get-ChildItem -Path '.build/' -Recurse -Include '*.ps1' -ErrorAction Ignore |
+        Get-ChildItem -Path '.build/' -Recurse -Include '*.build.ps1' -ErrorAction Ignore |
```
Comment (gaelcolas) [inline build.ps1:273]: As discussed, I'm just afraid that's a breaking change if it inadvertently make it to the template (say by copy/paste)...

**Review by gaelcolas — CHANGES_REQUESTED** [review]

File: build.ps1 (hunk 1 above)
Comment (johlju) [reply build.ps1]: Done. Move to `Sampler/scripts` and that is copied to root of built module. I rather would have wanted `.build/scripts` but then I would need to change build.ps1 again. So this was the best I could come up with.

**Comment by johlju** [comment]
@gaelcolas please see if this works.

**Comment by johlju** [comment]
@gaelcolas merging this. Assuming your thumbs up was approval for recent change. :smile:

**Pull request merged**